### DISCUSSION TITLE
Incorrect ptx device specification?

### DISCUSSION THREAD

**Discussion by zealbell** [beehive-lab/TornadoVM#221] [Q&A] [answered]
Is it normal for the _Total Number of Block Threads_ to be 2147483647 whenever my backend is `ptx` 
```bash
$ tornado --devices
WARNING: Using incubator modules: jdk.incubator.vector, jdk.incubator.foreign

Number of Tornado drivers: 1
Driver: PTX
  Total number of PTX devices  : 1
  Tornado device=1:0
        PTX -- PTX -- NVIDIA GeForce RTX 3080
                Global Memory Size: 11.8 GB
                Local Memory Size: 48.0 KB
                Workgroup Dimensions: 3
                Total Number of Block Threads: 2147483647 ##same value as  Integer.MAX_VALUE? 
                Max WorkGroup Configuration: [1024, 1024, 64]
                Device OpenCL C version: N/A
```

> meanwhile whenever my backend is `opencl` _Total Number of Block Threads_ is 1024  
> ```bash
> $ tornado --devices
> WARNING: Using incubator modules: jdk.incubator.vector, jdk.incubator.foreign
> 
> Number of Tornado drivers: 1
> Driver: OpenCL
>   Total number of OpenCL devices  : 1
>   Tornado device=0:0
>         OPENCL --  [NVIDIA CUDA] -- NVIDIA GeForce RTX 3080
>                 Global Memory Size: 11.8 GB
>                 Local Memory Size: 48.0 KB
>                 Workgroup Dimensions: 3
>                 Total Number of Block Threads: 1024
>                 Max WorkGroup Configuration: [1024, 1024, 64]
>                 Device OpenCL C version: OpenCL C 1.2
> 
> ```

**NVIDIA-SMI 520.61.05    Driver Version: 520.61.05    CUDA Version: 11.8**

**Comment by jjfumero** [comment, marked as answer]
Thanks for the report. This seems a bug in the TornadoVM Runtime. Can you please open an issue with this report?

**Comment by stratika** [comment]
Hi, I have just had a look and it seems that the `tornado --devices` reports the value that is being reported by the driver, and particularly using the `CU_DEVICE_ATTRIBUTE_MAX_GRID_DIM_X` [1]. In this case, the driver returns `(2^31 -1)` as the **_"maximum x-dimension of a grid of thread blocks"_** [2]. This is the phrasing used in the Nvidia documentation. For y and z dimensions this number is `65535`. So, I am not sure that this is a mistake. 

1. Do you anticipate to see any particular number different from the current one, based on your driver? 
2. Or did the "Total Number of Block Threads" confuse you?

Regarding the second question, we tried to have a common reporting field for all backends (OpenCL, PTX, SPIR-V), so it can create confusion.

[1] https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__DEVICE.html#group__CUDA__DEVICE_1g9c3e1414f0ad901d3278a4d6645fc266
[2] https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#features-and-technical-specifications

**Reply by jjfumero** [reply]
Let's continue the discussion in the issue #222
